oktaawscli/aws_auth.py

```diff
@@ -76,12 +76,18 @@ def choose_aws_role(self, assertion):
     def get_sts_token(self, role_arn, principal_arn, assertion, duration):
         """ Gets a token from AWS STS """
 
-        # Connect to the GovCloud STS endpoint if a GovCloud ARN is found.
-        arn_region = principal_arn.split(':')[1]
-        if arn_region == 'aws-us-gov':
-            sts = boto3.client('sts', region_name='us-gov-west-1')
-        else:
-            sts = boto3.client('sts', region_name=self.region)
+        try:
+            profile = os.environ.pop('AWS_PROFILE', None)
+            # Connect to the GovCloud STS endpoint if a GovCloud ARN is found.
+            arn_region = principal_arn.split(':')[1]
+            if arn_region == 'aws-us-gov':
+                sts = boto3.client('sts', region_name='us-gov-west-1')
```

Comment:
Nitpick, do you think we can get rid of this? Will we ever run into this case?

Reply:
Meh, you never know.